### ISSUE TITLE
🐛 [cloud_firestore] This is a most likely a transient condition and may be corrected by retrying with a backoff

### ISSUE THREAD

**Issue by 97loser** [FirebaseExtended/flutterfire#4139]
I'm using firebase in my project and i got some issues mentioned below - 

  [cloud_firestore/unavailable] The service is currently unavailable.
       This is a most likely a transient condition and may be corrected by retrying with a backoff.

I'm using firebase lib version : 

     firebase_auth: ^0.18.0+1
     google_sign_in: ^4.5.3
     cloud_firestore: ^0.14.3
     firebase_core: ^0.5.2
     firebase_messaging: ^7.0.3
     firebase_storage: ^4.0.0

   flutter run --verbose - result : 

         [ +208 ms] executing: [/home/loser97/Desktop/application/flutter/flutter_linux/flutter/] git -c
    log.showSignature=false log -n 1 --pretty=format:%H
    [ +472 ms[        ] f30b7f4db93ee747cd727df747941a28ead25ff5log -n 1 --pretty=format:%H
    [   +1 ms] executing: [/home/loser97/Desktop/application/flutter/flutter_linux/flutter/] git tag --points-at HEAD
    [+1208 ms] Exit code 0 from: git tag --points-at HEAD
    [        ] 1.22.0-12.4.pre
               1.22.1
    [  +14 ms] executing: [/home/loser97/Desktop/application/flutter/flutter_linux/flutter/] git rev-parse --abbrev-ref
    --symbolic @{u}
    [  +38 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
    [        ] origin/stable
    [        ] executing: [/home/loser97/Desktop/application/flutter/flutter_linux/flutter/] git ls-remote --get-url
    origin
    [   +9 ms] Exit code 0 from: git ls-remote --get-url origin
    [        ] https://github.com/flutter/flutter.git
    [  +83 ms] executing: [/home/loser97/Desktop/application/flutter/flutter_linux/flutter/] git rev-parse --abbrev-ref
    HEAD
    [   +6 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
    [        ] stable
    [ +157 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
    [  +20 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
    [ +162 ms] executing: /home/loser97/Android/Sdk/platform-tools/adb devices -l
    [  +52 ms] List of devices attached
    [   +3 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
    [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
    [        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
    [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
    [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
    [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
    [ +158 ms] No supported devices connected.
    [  +14 ms] "flutter run" took 502ms.
    [  +28 ms] 
               #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
               #1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:353:7)
               <asynchronous suspension>
               #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:933:11)
               <asynchronous suspension>
               #3      FlutterCommand.run.<anonymous closure>
    (package:flutter_tools/src/runner/flutter_command.dart:836:33)
               <asynchronous suspension>
               #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart)
               #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
               #6      _rootRun (dart:async/zone.dart:1190:13)
               #7      _CustomZone.run (dart:async/zone.dart:1093:19)
               #8      _runZoned (dart:async/zone.dart:1630:10)
               #9      runZoned (dart:async/zone.dart:1550:10)
               #10     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
               #11     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:825:20)
               #12     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
               #13     FlutterCommandRunner.runCommand.<anonymous closure>
               (package:flutter_tools/src/runner/flutter_command_runner.dart:335:21)
               #14     _rootRunUnary (dart:async/zone.dart:1198:47)
               #15     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
               #16     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
               #17     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
               #18     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
               #19     Future._completeWithValue (dart:async/future_impl.dart:529:5)
               #20     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
               #21     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
               #22     FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart)
               #23     _rootRunUnary (dart:async/zone.dart:1198:47)
               #24     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
               #25     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
               #26     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
               #27     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
               #28     Future._completeWithValue (dart:async/future_impl.dart:529:5)
               #29     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
               #30     _rootRun (dart:async/zone.dart:1190:13)
               #31     _CustomZone.run (dart:async/zone.dart:1093:19)
               #32     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
               #33     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
               #34     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
               #35     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
               #36     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
               #37     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
               
               
    [ +258 ms] ensureAnalyticsSent: 252ms
    [   +3 ms] Running shutdown hooks
    [        ] Shutdown hooks complete
    [        ] exiting with code 1

please share your answer.

**Comment by markusaksli-nc** [comment]
Hi @97loser 
Your `flutter run -v` ends with `[ +158 ms] No supported devices connected.`.
Could you please share the logs of `flutter run -v` where you encounter the mentioner error?
Could you also please provide your `flutter doctor -v` and `flutter pub deps -- --style=compact`?
Thank you

**Comment by ElaineSchwner** [comment]
Please help. Since June with this issue.
Check this https://github.com/FirebaseExtended/flutterfire/issues/2753 and this https://github.com/FirebaseExtended/flutterfire/issues/3521

**Comment by mdalihusain** [comment]
I also encountered this problem, but the issue has been resolved a few days back on one of my devices.
@Ehesp I would like to confirm that this issue was not limited to FlutterFire, native Android Firebase Apps were not working as well.
I was in contact with Firebase Support, they told me that there are some issues with ISPs on which they are working to get them resolved.
For me, things are working fine now, but once in a couple of weeks, I still face this issue for 5-10 mins. And, I have noticed that when this issue occurs, even YouTube takes a long time in loading.
